packages/suite-desktop/pages/_app.tsx

```diff
@@ -50,10 +58,10 @@ class TrezorSuiteApp extends App<Props> {
                             <Tor />
                             <OnlineStatus />
                             <IntlProvider>
-                                <DesktopUpdater />
+                                <DesktopUpdater setIsUpdateVisible={this.setIsUpdateVisible} />
                                 <Router />
                                 <ToastContainer />
-                                <Preloader>
+                                <Preloader hideModals={this.state.isUpdateVisible}>
```

Comment:
```suggestion
                                this.state.isUpdateVisible ? <Component {...pageProps} /> : <Preloader><Component {...pageProps} /></Preloader>
```
wouldn't it be better like this?
change will apply only to this particular file instead of introducing new hideModals prop

Reply:
`Preloader` dispatches actions (init) and wraps `children` in the `SuiteLayout` component, new prop seemed reasonable to me.

Reply:
the only children here is `<Component {...pageProps} />` the rest of possible render configurations (modals) in `Preloader` are disabled by new `hideModals` prop.
the only reason why `Preloader`  is rendered in this case is to render children. so why not to skip rendering `Preloader` and his logic at all? 
profit:
you dont have to add nothing into `Preloader`, and the whole "display priority" condition is where it's declared/used (_app.tsx)

Reply:
nope, i'm wrong and it shouldn't be done like that. `Preloader` should be mounted once and never unmounted because of init/loading/error useEffect logic. so i guess we need to do it with the prop